tasks/seal/task_synth_proofs.go

```diff
@@ -0,0 +1,203 @@
+		return true, nil
+	}
+
+	sealed, err := cid.Parse(sectorParams.SealedCID)
+	if err != nil {
+		return false, xerrors.Errorf("failed to parse sealed cid: %w", err)
+	}
+
+	unsealed, err := cid.Parse(sectorParams.UnsealedCID)
+	if err != nil {
+		return false, xerrors.Errorf("failed to parse unsealed cid: %w", err)
+	}
+
+	sref := storiface.SectorRef{
+		ID: abi.SectorID{
+			Miner:  abi.ActorID(sectorParams.SpID),
+			Number: abi.SectorNumber(sectorParams.SectorNumber),
+		},
+		ProofType: sectorParams.RegSealProof,
+	}
+
+	dealData, err := dealdata.DealDataSDRPoRep(ctx, s.db, s.sc, sectorParams.SpID, sectorParams.SectorNumber, sectorParams.RegSealProof, true)
+	if err != nil {
+		return false, xerrors.Errorf("getting deal data: %w", err)
+	}
+
+	err = s.sc.SyntheticProofs(ctx, &taskID, sref, sealed, unsealed, sectorParams.TicketValue, dealData.PieceInfos)
+	if err != nil {
+		serr := resetSectorSealingState(ctx, sectorParams.SpID, sectorParams.SectorNumber, err, s.db)
+		if serr != nil {
+			return false, xerrors.Errorf("generating synthetic proofs: %w", err)
+		}
+	}
+
+	err = s.markFinished(ctx, sectorParams.SpID, sectorParams.SectorNumber)
+	if err != nil {
+		return false, err
+	}
+
+	return true, nil
+}
+
+func resetSectorSealingState(ctx context.Context, spid, secNum int64, err error, db *harmonydb.DB) error {
+	if err != nil {
+		if strings.Contains(err.Error(), "checking PreCommit") {
+			n, serr := db.Exec(ctx, `UPDATE sectors_sdr_pipeline 
+						SET after_tree_d = false, tree_d_cid = NULL, after_tree_r = false, after_tree_c = false, task_id_tree_r = NULL, task_id_tree_c = NULL,
+						    after_synth = false, task_id_synth = null
+						WHERE sp_id = $1 AND sector_number = $2`, spid, secNum)
+			if serr != nil {
+				return xerrors.Errorf("store sdr-trees failure: updating pipeline: Original error %w: DB error %w", err, serr)
+			}
+			if n != 1 {
+				return xerrors.Errorf("store sdr-trees failure: Original error %w: updated %d rows", err, n)
+			}
+		}
+		return err
+	}
+	return nil
+}
+
+func (s *SyntheticProofTask) markFinished(ctx context.Context, spid, sector int64) error {
+	n, err := s.db.Exec(ctx, `UPDATE sectors_sdr_pipeline SET after_synth = true, task_id_synth = NULL 
+                            WHERE sp_id = $1 AND sector_number = $2`,
+		spid, sector)
+	if err != nil {
+		return xerrors.Errorf("store SyntheticProofs success: updating pipeline: %w", err)
+	}
+	if n != 1 {
+		return xerrors.Errorf("store SyntheticProofs success: updated %d rows", n)
+	}
+	return nil
+}
+
+func (s *SyntheticProofTask) CanAccept(ids []harmonytask.TaskID, engine *harmonytask.TaskEngine) (*harmonytask.TaskID, error) {
+	id := ids[0]
+	return &id, nil
+}
+
+func (s *SyntheticProofTask) TypeDetails() harmonytask.TaskTypeDetails {
+	ssize := abi.SectorSize(32 << 30)
+	ram := uint64(8 << 30)
+	if IsDevnet {
+		ssize = abi.SectorSize(2 << 20)
+		ram = uint64(1 << 30)
+	}
+
+	res := harmonytask.TaskTypeDetails{
+		Max:  50,
```

Comment:
Would add this max to the config, also default lower, probably 16

Reply:
Added to config with new default value as other don't have one either.